src/Native/Unix/System.Native/pal_networking.c

```diff
@@ -1255,6 +1256,29 @@ int32_t SystemNative_Accept(intptr_t socket, uint8_t* socketAddress, int32_t* so
     while ((accepted = accept4(fd, (struct sockaddr*)socketAddress, &addrLen, SOCK_CLOEXEC)) < 0 && errno == EINTR);
 #else
     while ((accepted = accept(fd, (struct sockaddr*)socketAddress, &addrLen)) < 0 && errno == EINTR);
+#if defined(FD_CLOEXEC)
+    // macOS does not have accept4 but it can set _CLOEXEC on descriptor.
+    // Unlike accept4 it is not atomic and the fd can leak child process.
```

Comment:
Nit: "the fd can leak child process" => "the fd can leak into the child process"